minify-html/mod.ts

```diff
@@ -0,0 +1,97 @@
+  /**
+   * Default options for minify-html library
+   * Opened an issue with a request of adding an export for the options type definitions so we can use `minifyOptions: Partial<CFG>`.
+   * https://github.com/wilsonzlin/minify-html/issues/97
+   * until then we define them manually.
+   */
+  options: {
+    /** Do not minify DOCTYPEs. Minified DOCTYPEs may not be spec compliant. */
+    do_not_minify_doctype?: boolean;
+    /** Ensure all unquoted attribute values in the output do not contain any characters prohibited by the WHATWG specification. */
+    ensure_spec_compliant_unquoted_attribute_values?: boolean;
+    /** Do not omit closing tags when possible. */
+    keep_closing_tags?: boolean;
+    /** Do not omit `<html>` and `<head>` opening tags when they don't have attributes. */
+    keep_html_and_head_opening_tags?: boolean;
+    /** Keep spaces between attributes when possible to conform to HTML standards. */
+    keep_spaces_between_attributes?: boolean;
+    /** Keep all comments. */
+    keep_comments?: boolean;
+    /** If enabled, content in `<script>` tags with a JS or no [MIME type](https://mimesniff.spec.whatwg.org/#javascript-mime-type) will be minified using [minify-js](https://github.com/wilsonzlin/minify-js). */
+    minify_js?: boolean;
+    /** If enabled, CSS in `<style>` tags and `style` attributes will be minified. */
+    minify_css?: boolean;
+    /** Remove all bangs. */
+    remove_bangs?: boolean;
+    /** Remove all processing_instructions. */
+    remove_processing_instructions?: boolean;
+  };
+}
+
+// Default options
+export const defaults: Options = {
+  extensions: [".html", ".css", ".js"],
+  options: {
+    do_not_minify_doctype: false,
+    ensure_spec_compliant_unquoted_attribute_values: false,
+    keep_closing_tags: false,
+    keep_html_and_head_opening_tags: false,
+    keep_spaces_between_attributes: false,
+    keep_comments: false,
+    minify_js: false,
+    minify_css: false,
+    remove_bangs: false,
+    remove_processing_instructions: false,
+  },
+};
+
+// Init minify-html
+await init();
+
+/** A plugin to minify HTML, CSS & JavaScript files */
+export default function (userOptions?: Partial<Options>) {
+  const options = merge(defaults, userOptions);
+
+  const { extensions } = options;
+
+  // Validate supported file extensions
+  if (extensions.some((ext) => ![".html", ".css", ".js"].includes(ext))) {
+    throw new Exception("Unsupported extensions configuration.", {
+      name: "Plugin minify-html",
+      extensions,
+    });
+  }
+
+  // When `extensions` includes `.css` and `.js`, minify-html will minify CSS and JavaScript files, even when `minify_css` and `minify_js` defaults set to `false`, since lume passes them to process
+  // To make this work with inline `<style>` tags in HTML files, following is necessary when `minify_css` is not set to `true`
+  const minifyOptions = { ...options.options };
+
+  if (extensions.includes(".css")) {
+    minifyOptions.minify_css = true;
```

Comment:
I don't understand this.

The `extensions` option is to configure the pages to minify (html, css and js pages). `options.minify_css` and `options.minify_js` configure the html minifier to minify inline css and js pages. They are different things that shouldn't be mixed.

Maybe this plugins should be configured by default to minify only `.html` extensions, with `minify_js` and`minify_css` to true (to minify also the inlined styles and scripts). But don't minify `.css` and `.js` files by default (to don't double-minify these files if the parcel_css or terser/esbuild plugins are being used).

Reply:
> Maybe this plugins should be configured by default to minify only .html extensions

Yeah, this was my actuall idea, this settings I had before I made the changes. I just thought it was a good idea to set `minify_js` and `minify_css` to `false` and automatically to `true`, if the user adds the extensions. So we talked past each other.

I agree with you and will apply the changes.